### PR TITLE
staging

### PR DESCRIPTION
**Pull request by ff6347**
- feat(http): adds batch posting of records
- fix(deps): update react monorepo to v18
- fix(http): fixes failing logic and adds tests
- fix(HTTP): Remove max number of records

- chore: Remove console.log
- chore(date-fns): move back to dev deps
- chore(deps): update dependency loadimpact/k6 to v0.37.0
- chore(deps): update featify, pino and pino-pretty
- chore(deps): update node.js to v14.19.1
- chore(release): 3.1.0-rc.4 [skip ci]
- chore(release): 3.1.0-rc.3 [skip ci]
- chore(release): 3.2.0-rc.2 [skip ci]
- chore(release): 3.2.0-rc.1 [skip ci]

- docs: Update rest calls [skip ci]
- docs(http-client): Adds delete records request

- test: Remove call to maxRecordsLength
- test: Remove only from test
- test(http): adds tests for batch inserting records
- test(tools): adds more launch configs

### CODE REVIEW

**Review by vogelino — APPROVED** [review]
Picobello! 👌

File: src/integrations/http.test.ts [hunk 1]
```diff
@@ -301,7 +316,7 @@ describe("tests for the http integration", () => {
     // end boilerplate
   });
 
-  test("should reject recorded_at if date-time cannot be parsed as date", async () => {
+  test.only("should reject recorded_at if date-time cannot be parsed as date", async () => {
```
Comment (ff6347) [inline src/integrations/http.test.ts:319]: remove only

**Review by ff6347 — COMMENTED** [review]
just one tiny thing

File: dev-tools/dev-client/package.json [hunk 2]
```diff
@@ -10,8 +10,8 @@
   },
   "dependencies": {
     "@supabase/supabase-js": "1.21.3",
-    "react": "17.0.2",
-    "react-dom": "17.0.2"
+    "react": "18.0.0",
```
Comment (vogelino) [inline dev-tools/dev-client/package.json:13]: React is necessary for the API?

**Review by vogelino — APPROVED** [review]
Perfect! Thx @ff6347 !

File: dev-tools/dev-client/package.json (hunk 2 above)
Comment (ff6347) [reply dev-tools/dev-client/package.json:13]: Nope. It is in the dev tools folder

**Pull request merged**